faststreams/inputs.nim

```diff
@@ -875,15 +875,24 @@ when fsAsyncSupport:
 template useHeapMem(_: Natural) =
   var buffer: seq[byte]
 
-  template allocMem(n: Natural): ptr byte =
-    buffer.setLen(n)
-    addr buffer[0]
+  when (NimMajor, NimMinor) > (1, 6):
+    template allocMem(n: Natural): ptr byte {.redefine.} =
+      buffer.setLen(n)
+      addr buffer[0]
+  else:
+    template allocMem(n: Natural): ptr byte =
+      buffer.setLen(n)
+      addr buffer[0]
 
 template useStackMem(n: static Natural) =
   var buffer: array[n + 1, byte]
 
-  template allocMem(_: Natural): ptr byte =
-    addr buffer[0]
+  when (NimMajor, NimMinor) > (1, 6):
+    template allocMem(_: Natural): ptr byte {.redefine.} =
```

Comment:
unfortunate

Reply:
Agree, but at least easy to find when eventually 1.6 support can be cleanly extirpated from `nim-faststreams`.

Reply:
https://github.com/status-im/nim-faststreams/pull/46 fixes the multiple-reads-per-scope issue which showed up in https://github.com/status-im/nimbus-eth2/pull/5123 and also does it in a way that's not dependent on Nim version anymore.